pkg/adapter/kubernetesevents/adapter_test.go

```diff
@@ -0,0 +1,186 @@
+		},
+		// Kubernetes event adapter does not care if the sink accepts the event or not.
+		"rejected": {
+			sink:         sinkRejected,
+			expectedBody: `{"metadata":{"uid":"ABC","creationTimestamp":"%s"},"involvedObject":{"kind":"Kind","namespace":"Namespace","name":"Name","apiVersion":"api/version","resourceVersion":"v1test1","fieldPath":"field"},"source":{},"firstTimestamp":null,"lastTimestamp":null,"eventTime":null,"reportingComponent":"","reportingInstance":""}`,
+		},
+	}
+	for n, tc := range testCases {
+		t.Run(n, func(t *testing.T) {
+			h := &fakeHandler{
+				handler: tc.sink,
+			}
+			sinkServer := httptest.NewServer(h)
+			defer sinkServer.Close()
+
+			a := &Adapter{
+				SinkURI:   sinkServer.URL,
+				Namespace: "default",
+				ctx:       context.TODO(),
+			}
+
+			uid := types.UID("ABC")
+
+			ref := corev1.ObjectReference{
+				Kind:            "Kind",
+				Namespace:       "Namespace",
+				Name:            "Name",
+				APIVersion:      "api/version",
+				ResourceVersion: "v1test1",
+				FieldPath:       "field",
+			}
+
+			now := time.Now()
+
+			event := &corev1.Event{
+				ObjectMeta: metav1.ObjectMeta{
+					UID:               uid,
+					CreationTimestamp: metav1.Time{Time: now},
+				},
+				InvolvedObject: ref,
+			}
+
+			a.updateEvent(nil, event)
+
+			expected := fmt.Sprintf(tc.expectedBody, now.UTC().Format(time.RFC3339))
+			got := string(h.body)
+			if expected != got {
+				t.Errorf("expected message sent to be %v, got %v", expected, got)
+			}
+		})
+	}
+}
+
+func TestUpdateEvent_NonExistentSink(t *testing.T) {
+	a := &Adapter{
+		SinkURI:   "http://localhost:50/",
+		Namespace: "default",
+		ctx:       context.TODO(),
+	}
+
+	uid := types.UID("ABC")
+
+	ref := corev1.ObjectReference{
+		Kind:            "Kind",
+		Namespace:       "Namespace",
+		Name:            "Name",
+		APIVersion:      "api/version",
+		ResourceVersion: "v1test1",
+		FieldPath:       "field",
+	}
+
+	now := time.Now()
+
+	event := &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:               uid,
+			CreationTimestamp: metav1.Time{Time: now},
+		},
+		InvolvedObject: ref,
+	}
+
+	a.updateEvent(nil, event)
+
+	t.Log("should get here without crashing.")
+}
+
+func TestPostMessage_InvalidSink(t *testing.T) {
+	a := &Adapter{
+		SinkURI:   "@#$@",
+		Namespace: "default",
+		ctx:       context.TODO(),
+	}
+
+	uid := types.UID("ABC")
+
+	ref := corev1.ObjectReference{
+		Kind:            "Kind",
+		Namespace:       "Namespace",
+		Name:            "Name",
+		APIVersion:      "api/version",
+		ResourceVersion: "v1test1",
+		FieldPath:       "field",
+	}
+
+	now := time.Now()
+
+	event := &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:               uid,
+			CreationTimestamp: metav1.Time{Time: now},
+		},
+		InvolvedObject: ref,
+	}
+
+	err := a.postMessage(event)
+
+	if err == nil {
```

Comment:
Can we be a bit more selective on what error we're looking for? A different error might cause this to pass incorrectly.